thorlcr/thorutil/thormisc.cpp

```diff
@@ -1337,5 +1363,40 @@ IPerfMonHook *createThorMemStatsPerfMonHook(CJobBase &job, int maxLevel, IPerfMo
     return new CPerfMonHook(job, maxLevel, chain);
 }
 
-
 const StatisticsMapping spillStatistics(StTimeSpillElapsed, StTimeSortElapsed, StNumSpills, StSizeSpillFile, StKindNone);
+
+bool isOOMException(IException *_e)
+{
+    if (_e)
+    {
+        IThorException *e = QUERYINTERFACE(_e, IThorException);
+        IException *oe = e && e->queryOriginalException() ? e->queryOriginalException() : _e;
+        int ecode = oe->errorCode();
```

Comment:
In theory yes. The original exception can be to any, so could differ from owner.
In practice, not so far, setOriginalException() is always being called from a wrapper function which sets parent error code to code of original.

If there's a origin exception, I want to make sure I'm checking it's code.